### ISSUE TITLE
custom_lint does not apply lints from transitive packages

### ISSUE THREAD

**Issue by ChaserVasya** [invertase/dart_custom_lint#372]
### app_lint package
```yaml
# packages/app_lint/lib/analysis_options.yaml
analyzer:
  plugins:
    - custom_lint
...
```

```yaml
# packages/app_lint/pubspec.yaml
...
dependencies:
  altive_lints: ^1.22.0
  custom_lint: ^0.7.0
...
```

### main package

```yaml
# analysis_options.yaml
dev_dependencies:
  app_lint:
    path: packages/app_lint
```

```dart
// lib/main.dart

void main(){
  DateTime.now();
}
```

## Expected behavior
`dart run custom_lint` outputs 
```console
Analyzing...                           0.0s

  lib/main.dart:9:3 • Avoid using DateTime.now(). Use a testable alternative like clock.now() or similar instead. • prefer_clock_now • INFO

1 issue found.
```

## Actual behavior
`dart run custom_lint` outputs `No issues found!`
BUT
If I write next:
```yaml
# analysis_options.yaml
dev_dependencies:
  app_lint:
    path: packages/app_lint
  altive_lints: ^1.22.0
  custom_lint: ^0.7.0
```
it outputs an issue

## Context 
dart info
```console
- Dart 3.8.1 (stable) (Wed May 28 00:47:25 2025 -0700) on "macos_arm64"
- on macos / Version 26.1 (Build 25B78)
- locale is ru-RU
```
custom lint: 0.7.6
